### PR TITLE
Refactor FTP functions to alllow better unit tests

### PR DESCRIPTION
**Pull request by KJone1**

### CODE REVIEW

File: ftp/ftp.go [hunk 1]
```diff
@@ -33,109 +30,117 @@ func Client() (*FtpClient, error) {
...
 }
 
-func Connect() (*FtpClient, error) {
-	config := config.FtpConfig()
+func Connect(URL string, userName string, passWord string) (*goftp.Client, error) {
 
 	ftpConfig := goftp.Config{
-		User:               config.FtpUsername,
-		Password:           config.FtpPassword,
+		User:               userName,
```
Comment (dorimon-1) [inline ftp/ftp.go:36]: wtf bro userName passWord?
username, password

File: ftp/ftp.go [hunk 2]
```diff
@@ -33,109 +30,117 @@ func Client() (*FtpClient, error) {
...
+
+		if err = client.Retrieve(remote_file, buffer); err != nil {
+			log.Error().Msgf("Failed to retreive file with error => %s", err)
+			continue
 		}
-    } 
+		filesPulled = append(filesPulled, local_file)
+
+		// // TODO: Move to seperate function becouse decopress not always needed when pulling e.g. voice station
+		// local_file := workingDir + "/" + file
```
Comment (dorimon-1) [inline ftp/ftp.go:77]: remove commented code

**Review by dorimon-1 — CHANGES_REQUESTED** [review]

File: ftp/ftp.go [hunk 3]
```diff
@@ -33,109 +29,118 @@ func Client() (*FtpClient, error) {
 	return client, nil
 }
 
-func Connect() (*FtpClient, error) {
-	config := config.FtpConfig()
+func Connect(URL string, Username string, Password string) (*goftp.Client, error) {
```
[the remark this answers is not in the document]
Comment (dorimon-1) [reply ftp/ftp.go]: camelCase is DavidYahin, but username and password are a single word

File: ftp/ftp.go (hunk 2 above)
Comment (KJone1) [reply ftp/ftp.go:77]: Are we going to separate this function into standalone operation only for images?

Comment (dorimon-1) [reply ftp/ftp.go:77]: we will have to do WithDecompression or not

**Review by dorimon-1 — APPROVED** [review]

**Pull request merged**